### PR TITLE
Feature/sql postgres

### PR DESCRIPTION
**Pull request by PinkhamSoftware**
What this does:
- Connect to postgres database
- Test database connection
- Add Docker images to run tests in circle ci
- Adjust CircleCi configs
- Add 2 test projects to circle steps
- Adjust Makefile for testing

What we will tackle in a different PR
- Tackle Migrations
- Creating Database
- Creating Table
- Inserting Test Data

### CODE REVIEW

File: .circleci/config-2-1.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
+version: 2.1
```
Comment (danielburnley) [inline .circleci/config-2-1.yml]: :eyes:

File: HomesEngland.Gateway.Test/PostgresConnectionFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        private IDatabaseConnectionFactory _classUnderTest = null;
+        public PostgresConnectionFactoryTests()
+        {
+            _classUnderTest = new PostgresDatabaseConnectionFactory();
+        }
+
+        [Test]
+        public void GivenValidConnectionString_ThenDoesntThrowExceptionWhenConnectingToDatabase()
+        {
+            //arrange
```
Comment (danielburnley) [inline HomesEngland.Gateway.Test/PostgresConnectionFactoryTests.cs]: I'd remove these comments as they don't particularly add anything

File: HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace HomesEngland.Gateway
+{
+    public class PostgresDatabaseConnectionFactory : IDatabaseConnectionFactory
+    {
+        public IDbConnection Create(string connectionString)
+        {
+
+            var uri = new Uri(connectionString);
+            //postgres://u1gkqgfiyblvhxa7:TbSMnjtd7EFm2BC1YObfeQrWXD5ftTQ4@rdsbroker-b85444a7-5561-4f75-81da-f8c82d5c5815.c7uewwm9qebj.eu-west-1.rds.amazonaws.com:5432/rdsbroker_b85444a7_5561_4f75_81da_f8c82d5c5815
```
Comment (danielburnley) [inline HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs]: :eyes: will want to remove this

Comment (PinkhamSoftware) [reply HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs]: Done

File: HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Data;
+using System.Linq;
+using Npgsql;
+
+namespace HomesEngland.Gateway
+{
+    public class PostgresDatabaseConnectionFactory : IDatabaseConnectionFactory
+    {
+        public IDbConnection Create(string connectionString)
```
Comment (danielburnley) [inline HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs]: I'd change `connectionString` to `DatabaseURL` as that's more descriptive of what we're taking

Once this is renamed you could extract the below out into a private method called `getConnectionStringFromURL` or the like to be clearer as to what's happening

File: HomesEngland.Gateway/SqlAssetReader.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
```
Comment (danielburnley) [inline HomesEngland.Gateway/SqlAssetReader.cs]: :eyes: I'd avoid putting this file in until it had tests - but given that's what you're working on next no big issue

File: HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs (hunk 4 above)
Comment (PinkhamSoftware) [reply HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs]: Yes, but this is also inheriting from what is a generic interface. The connection string is coming off the interface. I'm happy to change it, I think the question I have in mind is do you think connectionString or databaseUrl is more generic?

File: HomesEngland.Gateway.Test/PostgresConnectionFactoryTests.cs (hunk 2 above)
Comment (PinkhamSoftware) [reply HomesEngland.Gateway.Test/PostgresConnectionFactoryTests.cs]: Yes, I think that is a good point depending on who you are considering these comments are for. On my own personal projects I don't add these as I think they are unnecessary. However this is supposed to be a best practise project. A potential example for others to look at as to processes, methodologies, practises (testing), technologies etc. So whilst this adds little benefit from our perspective, someone may come across them and find that the Triple A standard for testing //arrange //act //assert helps them solidify their approach to Test Driven Development or unit testing in general.

File: HomesEngland.Gateway/SqlAssetReader.cs (hunk 5 above)
Comment (PinkhamSoftware) [reply HomesEngland.Gateway/SqlAssetReader.cs]: I agree it doesn't have tests surrounding it and isn't currently working so I'll remove it.

File: HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs (hunk 4 above)
Comment (PinkhamSoftware) [reply HomesEngland.Gateway/PostgresDatabaseConnectionFactory.cs]: Done

**Pull request merged**